akka-http-core/src/main/scala/akka/http/impl/engine/server/ServerTerminator.scala

```diff
@@ -77,6 +77,31 @@ private[http] final class MasterServerTerminator(log: LoggingAdapter) extends Se
     }
   }
 
+  /**
+   * Removes a previously registered per-connection terminator.
+   * Terminators must remove themselves like this once their respective connection is closed,
+   * otherwise they would leak and remain in the set indefinitely.
+   *
+   * If called with terminator that is not registered, false is returned but it is not an error.
```

Comment:
I don't think we actually fulfill this promise: we return 'false' only when the connection is not removed because the server (possibly including this connection) is already being terminated.

I'm not sure we need to either, it might be fine to just return `Unit`?

Reply:
True, good I'll simplify to be a simple `: Unit` return type